File: t/CS/ImageProcessingDemo/UI/Command/ResizeInterpolateDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools;
using Leadtools.ImageProcessing.Core;

namespace ImageProcessingDemo
{
   //This dialog is used to allow the user to specify values which
   //will be used for the ResizeInterpolateCommand

   public partial class ResizeInterpolateDialog : Form
   {
      private ResizeInterpolateCommand _ResizeInterpolateCommand;
      private RasterImage RasImage;

      public int ImageWidth;
      public int ImageHeight;

      public ResizeInterpolateDialog()
      {
         InitializeComponent();
         _ResizeInterpolateCommand = new ResizeInterpolateCommand();
         
         //Set command default values
         InitializeUI();
      }

      public ResizeInterpolateDialog(RasterImage TempImage)
      {
         InitializeComponent();
         _ResizeInterpolateCommand = new ResizeInterpolateCommand();
         RasImage = TempImage;
         InitializeUI();
      }

      public ResizeInterpolateCommand ResizeInterpolatecommand
      {
         get
         {
            //Update command values
            UpdateCommand();
            return _ResizeInterpolateCommand;
         }
         set
         {
            _ResizeInterpolateCommand = value;
            InitializeUI();
         }
      }

      private void InitializeUI()
      {

         string[] names;
         names = Enum.GetNames(typeof(ResizeInterpolateCommandType));
         foreach (string name in names)
         {
            _cmbInterpolation.Items.Add(name);
         }
         _cmbInterpolation.SelectedIndex = _cmbInterpolation.Items.IndexOf(_ResizeInterpolateCommand.ResizeType.ToString());

         ImageWidth = RasImage.Width;
         ImageHeight = RasImage.Height;

         _numWidth.Value = ImageWidth;
         _numHeight.Value = ImageHeight;
      }

      private void UpdateCommand()
      {
         ImageWidth = Convert.ToInt32(_numWidth.Value);
         ImageHeight = Convert.ToInt32(_numHeight.Value);

         _ResizeInterpolateCommand.Width = ImageWidth;
         _ResizeInterpolateCommand.Height = ImageHeight;
         _ResizeInterpolateCommand.ResizeType = TranslateDirection();
      }


      public ResizeInterpolateCommandType TranslateDirection()
      {
         switch (_cmbInterpolation.SelectedIndex)
         {
            case 0:
               return ResizeInterpolateCommandType.Normal;
            case 1:
               return ResizeInterpolateCommandType.Resample;
            case 2:
               return ResizeInterpolateCommandType.Bicubic;
            case 3:
               return ResizeInterpolateCommandType.Triangle;
            case 4:
               return ResizeInterpolateCommandType.Hermite;
            case 5:
               return ResizeInterpolateCommandType.Bell;
            case 6:
               return ResizeInterpolateCommandType.QuadraticBSpline;
            case 7:
               return ResizeInterpolateCommandType.CubicBSpline;
            case 8:
               return ResizeInterpolateCommandType.BoxFilter;
            case 9:
               return ResizeInterpolateCommandType.Lanczos;
            case 10:
               return ResizeInterpolateCommandType.Michell;
            case 11:
               return ResizeInterpolateCommandType.Cosine;
            case 12:
               return ResizeInterpolateCommandType.Catrom;
            case 13:
               return ResizeInterpolateCommandType.Quadratic;
            case 14:
               return ResizeInterpolateCommandType.CubicConvolution;
            case 15:
               return ResizeInterpolateCommandType.Bilinear;
            case 16:
               return ResizeInterpolateCommandType.Bresenham;
            default:
               return ResizeInterpolateCommandType.Normal;
         }
      }

      private void _btnOk_Click(object sender, EventArgs e)
      {
         UpdateCommand();
         this.DialogResult = DialogResult.OK;
         this.Close();
      }

      private void _btnCancel_Click(object sender, EventArgs e)
      {
         InitializeUI();
         this.DialogResult = DialogResult.Cancel;
      }
   }
}

File: t/VB/ImageProcessingDemo/UI/Command/ResizeInterpolateDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools
Imports Leadtools.ImageProcessing.Core

Namespace ImageProcessingDemo
   'This dialog is used to allow the user to specify values which
   'will be used for the ResizeInterpolateCommand

   Partial Public Class ResizeInterpolateDialog : Inherits Form
      Private _ResizeInterpolateCommand As ResizeInterpolateCommand
      Private RasImage As RasterImage

      Public ImageWidth As Integer
      Public ImageHeight As Integer

      Public Sub New()
         InitializeComponent()
         _ResizeInterpolateCommand = New ResizeInterpolateCommand()

         'Set command default values
         InitializeUI()
      End Sub

      Public Sub New(ByVal TempImage As RasterImage)
         InitializeComponent()
         _ResizeInterpolateCommand = New ResizeInterpolateCommand()
         RasImage = TempImage
         InitializeUI()
      End Sub

      Public Property ResizeInterpolatecommand() As ResizeInterpolateCommand
         Get
            'Update command values
            UpdateCommand()
            Return _ResizeInterpolateCommand
         End Get
         Set(ByVal value As ResizeInterpolateCommand)
            _ResizeInterpolateCommand = value
            InitializeUI()
         End Set
      End Property

      Private Sub InitializeUI()

         Dim names As String()
         names = System.Enum.GetNames(GetType(ResizeInterpolateCommandType))
         For Each name As String In names
            _cmbInterpolation.Items.Add(name)
         Next name
         _cmbInterpolation.SelectedIndex = _cmbInterpolation.Items.IndexOf(_ResizeInterpolateCommand.ResizeType.ToString())

         ImageWidth = RasImage.Width
         ImageHeight = RasImage.Height

         _numWidth.Value = ImageWidth
         _numHeight.Value = ImageHeight
      End Sub

      Private Sub UpdateCommand()
         ImageWidth = Convert.ToInt32(_numWidth.Value)
         ImageHeight = Convert.ToInt32(_numHeight.Value)

         _ResizeInterpolateCommand.Width = ImageWidth
         _ResizeInterpolateCommand.Height = ImageHeight
         _ResizeInterpolateCommand.ResizeType = TranslateDirection()
      End Sub


      Public Function TranslateDirection() As ResizeInterpolateCommandType
         Select Case _cmbInterpolation.SelectedIndex
            Case 0
               Return ResizeInterpolateCommandType.Normal
            Case 1
               Return ResizeInterpolateCommandType.Resample
            Case 2
               Return ResizeInterpolateCommandType.Bicubic
            Case 3
               Return ResizeInterpolateCommandType.Triangle
            Case 4
               Return ResizeInterpolateCommandType.Hermite
            Case 5
               Return ResizeInterpolateCommandType.Bell
            Case 6
               Return ResizeInterpolateCommandType.QuadraticBSpline
            Case 7
               Return ResizeInterpolateCommandType.CubicBSpline
            Case 8
               Return ResizeInterpolateCommandType.BoxFilter
            Case 9
               Return ResizeInterpolateCommandType.Lanczos
            Case 10
               Return ResizeInterpolateCommandType.Michell
            Case 11
               Return ResizeInterpolateCommandType.Cosine
            Case 12
               Return ResizeInterpolateCommandType.Catrom
            Case 13
               Return ResizeInterpolateCommandType.Quadratic
            Case 14
               Return ResizeInterpolateCommandType.CubicConvolution
            Case 15
               Return ResizeInterpolateCommandType.Bilinear
            Case 16
               Return ResizeInterpolateCommandType.Bresenham
            Case Else
               Return ResizeInterpolateCommandType.Normal
         End Select
      End Function

      Private Sub _btnOk_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnOk.Click
         UpdateCommand()
         Me.DialogResult = DialogResult.OK
         Me.Close()
      End Sub

      Private Sub _btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnCancel.Click
         InitializeUI()
         Me.DialogResult = DialogResult.Cancel
      End Sub
   End Class
End Namespace
